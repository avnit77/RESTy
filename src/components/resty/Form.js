import React from 'react';
import PropTypes from 'prop-types';

const Form = ({ onSubmit, onChange, method, url, textInput, username, password, token }) => {
  return (
    <form onSubmit={onSubmit}>
      <input type="text" placeholder="URL" value={url} name="url" onChange={onChange} />
      <section>
        <label htmlFor={name}><input id={name} type="radio" checked={method === 'GET'} name="method" value="GET" onChange={onChange}></input><span>GET</span></label>
        <label><input type="radio" checked={method === 'POST'} name="method" value="POST" onChange={onChange}></input><span>POST</span></label>
        <label><input type="radio" checked={method === 'PATCH'} name="method" value="PATCH" onChange={onChange}></input><span>PATCH</span></label>
        <label><input type="radio" checked={method === 'PUT'} name="method" value="PUT" onChange={onChange}></input><span>PUT</span></label>
        <label><input type="radio" checked={method === 'DELETE'} name="method" value="DELETE" onChange={onChange}></input><span>DELETE</span></label>
        <button type='submit'>GO</button>
        <button type='reset'>Clear All</button>
      </section>
      <section>
        <textarea disabled={method === 'GET' || method === 'DELETE'} placeholder='Raw JSON Body' name='textInput' value={textInput} onChange={onChange}></textarea>
      </section>
      <div>
        <p>Basic Authorization</p>
        <input className='username' type='text' name='username' placeholder='Username' value={username} onChange={onChange}></input>
        <input className='password' type='password' name='password' placeholder='Password' value={password} onChange={onChange}></input>
        <p>Bearer Token</p>
        <input className='text' type='text' name='token' placeholder='Bearer Token' value={token} onChange={onChange}></input>
      </div>
    </form>
  );
};

Form.propTypes = {
  method: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  textInput: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  username: PropTypes.string,
  password: PropTypes.string,
  token: PropTypes.string,
  onChange: PropTypes.func.isRequired
};

export default Form;

