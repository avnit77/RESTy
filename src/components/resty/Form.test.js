import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Quote component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Form/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
