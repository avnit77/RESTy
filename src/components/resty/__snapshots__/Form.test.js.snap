// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<form>
  <input
    name="url"
    placeholder="URL"
    type="text"
    value=""
  />
  <section>
    <label
      htmlFor=""
    >
      <input
        checked={false}
        id=""
        name="method"
        type="radio"
        value="GET"
      />
      <span>
        GET
      </span>
    </label>
    <label>
      <input
        checked={false}
        name="method"
        type="radio"
        value="POST"
      />
      <span>
        POST
      </span>
    </label>
    <label>
      <input
        checked={false}
        name="method"
        type="radio"
        value="PATCH"
      />
      <span>
        PATCH
      </span>
    </label>
    <label>
      <input
        checked={false}
        name="method"
        type="radio"
        value="PUT"
      />
      <span>
        PUT
      </span>
    </label>
    <label>
      <input
        checked={false}
        name="method"
        type="radio"
        value="DELETE"
      />
      <span>
        DELETE
      </span>
    </label>
    <button
      type="submit"
    >
      GO
    </button>
    <button
      type="reset"
    >
      Clear All
    </button>
  </section>
  <section>
    <textarea
      disabled={false}
      name="textInput"
      placeholder="Raw JSON Body"
    />
  </section>
  <div>
    <p>
      Basic Authorization
    </p>
    <input
      className="username"
      name="username"
      placeholder="Username"
      type="text"
      value=""
    />
    <input
      className="password"
      name="password"
      placeholder="Password"
      type="password"
      value=""
    />
    <p>
      Bearer Token
    </p>
    <input
      className="text"
      name="token"
      placeholder="Bearer Token"
      type="text"
      value=""
    />
  </div>
</form>
`;
